File: algorithms_and_data_structures/Projects/Project2/ola2.cpp
////Author:      Ryan Florida
////Project:     OLA2
////Due Date:    February 15, 2017
////Description: This program will implement two classes in order to maintain a
////             roster of students with associated information about those
////             students (c-number, grades, etc). In this particular file, the
////             user can label the roster however they please, choose their
////             desired input file, then they are prompted to enter a specific
////             student's c-number so as to see that student's information.
////             Once all of this is done, the entire roster (ie all of the
////             student information) is displayed for the user.
////
////             A note on the .h files: I did not include headers there,
////             except at the locations I declared methods because I did not
////             think it was reasonable to say I authored those files.


#include <iostream>
#include "Roster.h"
#include <string>

//Constant declared to provide an upper-bound on the number of students in a
//particular roster.
const int MAX = 30;

//0: Main.
int main()
{
    //Variable declarations.
    std::string c_number;
    Roster      adsRoster("CSCI 3110"); //Input string will determine roster
                                        //name.

    //Calling method to read in data file containing information about
    //students. Input string will determine input file.
    adsRoster.readStudentRecord("point.dat");

    //Prompting user to input student's c-number so as to display the desired
    //student's information.
    std::cout << "Please input a valid c-number: ";
    std::cin  >> c_number;
    std::cout << std::endl;
    adsRoster.printStudentInformation(c_number);

    //Prints the information of all students to standard out.
    adsRoster.printRoster();

    return 0;
}

File: theory_of_programming_languages/Project2/ola2.pl
#### Author:      Ryan Florida
#### Program:     OLA2
#### File:        ola2.pl
#### Date:        April 10, 2017
#### Description: This program will implement two classes in order to maintain
####              a roster of students with associated information about those
####              students (c-number, grades, etc). In this particular file, 
####              the user can label the roster however they please, choose 
####              their desired input file, then they are prompted to enter a 
####              specific student's c-number so as to see that student's 
####              information. Once all of this is done, the entire roster
####              (ie all of the student information) is displayed for the
####              user. Written in the Perl programming language.

use lib 'lib';
use Roster;
use warnings;
use strict;

#0: Main.
sub main
{
	#Clear terminal window to enhance asthetics.
	system('clear');
	
	#Declare instance of Roster class, namely a CSCI 3210 roster.
	my $theoryRoster = Roster->new('3210');

	#Prompt user for input file containing roster data.
	print "Please enter the roster file: ";
	my $inputFile = <STDIN>;

	#Read the input file into the roster object.
	$theoryRoster->readStudentRecord($inputFile);

	#Prompt the user to query about two students on the roster, then display
	#the appropriate information.
	for(my $i = 0; $i < 2; $i++)
	{
        print "Please input a valid c-number: ";
     	my $c_number = <STDIN>;
     	print "\n";
     	$theoryRoster->printStudentInformation($c_number);
    }

    #Print the entire roster.
    $theoryRoster->printRoster();

    return;
}

#Call main.
main();